### ISSUE TITLE
[1.0.2] It should not be possible to use weapons without enough weapon energy left in them.

### ISSUE THREAD

**Issue by Zieldak** [MegamixEngine/MegamixEngine#327]
It should not be possible to use weapons without enough weapon energy left in them. For example, it should be impossible to use the Water Shield if it only has 2 units of energy left.

**Comment by ginger-mochi** [comment]
I don't think this should be the default behaviour but this could be a good engine-config option.
